src/pbt/v2/project_config.py

```diff
@@ -68,6 +74,15 @@ class DataprocInfo(BaseModel):
     key_json: str
     location: str
 
+    def _bucket(self):
+        return self.bucket.replace("gs://", "").split("/")
+
+    def bare_bucket(self):
+        return self._bucket()[0]
+
+    def bare_path_prefix(self):
+        return "/".join(self._bucket()[1:])
```

Comment:
same methods in s3 as well. 

* We can refactor to some base class, or some util methods
* We can use urlparse instead of string manipulation, that way it'll not need more changes for synapse in future.